src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,141 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+
+    private static final int CORRECT_AGE = 50;
+    private static final int SMALLER_THAN_18_AGE = 2;
+    private static final int NEGATIVE_AGE = -100;
+    private static final String CORRECT_LOGIN = "CorrectLogin@gmail.login";
+    private static final String CORRECT_PASSWORD = "greatPassword12345";
+    private static final String SMALL_LENGTH_WORD = "small";
+
+    private User user;
+    private RegistrationServiceImpl registrationService;
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.PEOPLE.clear();
+    }
+
+    @Test
+    void register_addAndReturnCorrectUser_ok() {
+        user.setAge(CORRECT_AGE);
+        user.setLogin(CORRECT_LOGIN);
+        user.setPassword(CORRECT_PASSWORD);
+        User returnedUser = registrationService.register(user);
+        StorageDaoImpl storageDao = new StorageDaoImpl();
+        User actualAddedUser = storageDao.get(user.getLogin());
+        assertEquals(actualAddedUser, user);
+        assertEquals(returnedUser, user);
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        user = null;
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_allFieldsAreNull_notOk() {
+        user.setAge(null);
+        user.setLogin(null);
+        user.setPassword(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
```

Comment:
You should test only one invalid parameter in each test case. Let’s  remove this case